### PR TITLE
Added inference module

### PR DESCRIPTION
**Pull request by dellaert**
This fixes the fact that the inference module was forgotten when we split up the wrapper :-)

### CODE REVIEW

**Review by ProfFan — APPROVED** [review]
LGTM

**Comment by dellaert** [comment]
Thanks for the quick review, @ProfFan !

**Comment by varunagrawal** [comment]
I don't understand how this changes anything. It's not like we do `import gtsam.linear` right?

That being said, I am disliking the multiple #include pybind/stl.h. There should just be one and I'll do that next week since this has been on the wrapper agenda long enough :/

**Comment by varunagrawal** [comment]
Is VariableIndex the big update? Cool!

File: gtsam/inference/inference.i [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+// Headers for overloaded methods below, break hierarchy :-/
+#include <gtsam/linear/GaussianFactorGraph.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+#include <gtsam/symbolic/SymbolicFactorGraph.h>
+
+class VariableIndex {
+  // Standard Constructors and Named Constructors
+  VariableIndex();
+  // TODO: Templetize constructor when wrap supports it
```
Comment (varunagrawal) [inline gtsam/inference/inference.i:142]: FYI wrapper supports constructor templates.

**Pull request merged**